elastic/kibana
# [Index Management][Mappings editor] Allow multi-fields support for most field types

## Summary
Fixes https://github.com/elastic/kibana/issues/177822
Fixes https://github.com/elastic/kibana/issues/112879

This PR allows adding multi-fields to most mapping fields except the following list
> flattened
> aggregate_metric_double
> join
> constant_keyword
> shape
> geo_shape
> point

This fixes a bug with mutli-fields being dropped from a field when the UI was not supporting mutli-fields for that field type. 

### How to test 
Adding multi-fields
1. Navigate to Index Management -> Index templates or Component templates tab
2. Start creating a template and get to the Mappings step in the wizard
3. Try out different type fields and check that it's possible to add mutli-fields

Not dropping multi-fields
1. Create an index template 
```
PUT _index_template/test
{
  "template": {
    "mappings": {
      "properties": {
        "command_line": {
          "fields": {
            "text": {
              "type": "match_only_text"
            }
          },
          "type": "wildcard"
        }
      }
    }
  },
  "index_patterns": [
    "test"
  ],
  "data_stream": {},
  "allow_auto_create": false
}
```
2. Navigate to Index Management -> Index templates
3. Find the created template, open it and save without changes
4. Check that the nested multi-fields are still present in the field 
`GET _index_template/test `

### Screenshots 
<img width="907" alt="Screenshot 2024-03-07 at 17 29 03" src="https://github.com/elastic/kibana/assets/6585477/9a16d24d-b68d-479d-928b-58a5ec5c9147">
<img width="904" alt="Screenshot 2024-03-07 at 17 30 27" src="https://github.com/elastic/kibana/assets/6585477/121d5946-7bad-4ffb-afa3-c33c81b793a6">




### Checklist

Delete any items that are not applicable to this PR.



### Risk Matrix

Delete this section if it is not applicable to this PR.

Before closing this PR, invite QA, stakeholders, and other developers to identify risks that should be tested prior to the change/feature release.

When forming the risk matrix, consider some of the following examples and how they may potentially impact the change:

| Risk                      | Probability | Severity | Mitigation/Notes        |
|---------------------------|-------------|----------|-------------------------|
| Multiple Spaces&mdash;unexpected behavior in non-default Kibana Space. | Low | High | Integration tests will verify that all features are still supported in non-default Kibana Space and when user switches between spaces. |
| Multiple nodes&mdash;Elasticsearch polling might have race conditions when multiple Kibana nodes are polling for the same tasks. | High | Low | Tasks are idempotent, so executing them multiple times will not result in logical error, but will degrade performance. To test for this case we add plenty of unit tests around this logic and document manual testing procedure. |
| Code should gracefully handle cases when feature X or plugin Y are disabled. | Medium | High | Unit tests will verify that any feature flag or plugin combination still results in our service operational. |
| [See more potential risk examples](https://github.com/elastic/kibana/blob/main/RISK_MATRIX.mdx) |


### For maintainers
